pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -742,3 +772,74 @@ func handleGlobalConfigDelete(ctxArg interface{}, key string,
 	*ctx.globalConfig = *types.DefaultConfigItemValueMap()
 	log.Functionf("handleGlobalConfigDelete done for %s", key)
 }
+
+func handleZedAgentStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	handleZedAgentStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleZedAgentStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	handleZedAgentStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleZedAgentStatusImpl(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctxPtr := ctxArg.(*zedmanagerContext)
+	status := statusArg.(types.ZedAgentStatus)
+	if ctxPtr.currentProfile != status.CurrentProfile {
+		log.Noticef("handleZedAgentStatusImpl: CurrentProfile changed from %s to %s",
+			ctxPtr.currentProfile, status.CurrentProfile)
+		ctxPtr.currentProfile = status.CurrentProfile
+		updateBasedOnProfile(ctxPtr)
+	}
+	log.Functionf("handleZedAgentStatusImpl(%s) done", key)
+}
+
+// updateBasedOnProfile check all app instances with currentProfile, set EffectiveActivate and update app instances
+func updateBasedOnProfile(ctx *zedmanagerContext) {
+	pub := ctx.subAppInstanceConfig
+	items := pub.GetAll()
+	for _, c := range items {
+		config := c.(types.AppInstanceConfig)
+		effectiveActivate := effectiveActivateCurrentProfile(config, ctx.currentProfile)
+		status := lookupAppInstanceStatus(ctx, config.Key())
+		if status != nil {
+			if status.EffectiveActivate != effectiveActivate {
+				log.Functionf("updateBasedOnProfile: change activate state for %s from %t to %t",
+					config.Key(), status.EffectiveActivate, effectiveActivate)
+				status.EffectiveActivate = effectiveActivate
+				changed := doUpdate(ctx, config, status)
+				if changed {
```

Comment:
Since you already changed EffectiveActivate you always need to publish here. And in some case doUpdate might return false if it didn't cause any change.